### PR TITLE
feat: add 羊城晚报金羊网

### PR DESCRIPTION
**Pull request by TimWu007**
## 该 PR 相关 Issue / Involved issue

Close #

## 完整路由地址 / Example for the proposed route(s)

```routes
/ycwb/1
```

## 新 RSS 检查列表 / New RSS Script Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/ycwb/index.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    const { data: response } = await got(currentUrl);
+
+    const list = response.artiles.map((item) => ({
+        title: item.TITLE,
+        description: art(path.join(__dirname, 'templates/description.art'), {
+            thumb: item.PICLINKS,
+            description: item.ABSTRACT,
+        }),
+        pubDate: item.PUBTIME,
```
Comment (TonyRL) [inline lib/v2/ycwb/index.js:20]: Timezone adjustment needed

File: lib/v2/ycwb/index.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
+const got = require('@/utils/got');
+const cheerio = require('cheerio');
+const { art } = require('@/utils/render');
+const path = require('path');
+
+module.exports = async (ctx) => {
+    const node = ctx.params.node ?? 1;
+    const page = ctx.params.page ?? 1;
```
Comment (TonyRL) [inline lib/v2/ycwb/index.js:8]: Only the first page is enough

File: lib/v2/ycwb/index.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    const pageSize = ctx.params.pagesize ?? 15;
```
Comment (TonyRL) [inline lib/v2/ycwb/index.js:9]: Please use the common query parameter `ctx.query.limit` instead.
docs: https://docs.rsshub.app/parameter.html

File: lib/v2/ycwb/index.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+                nodeName = item.nodeName;
+                return item;
+            })
+        )
+    );
+
+    ctx.state.data = {
+        title: `羊城晚报金羊网 - ${nodeName}`,
+        link: 'http://www.ycwb.com',
```
Comment (TonyRL) [inline lib/v2/ycwb/index.js:70]: Maybe use the last element of $('.path a') in article page instead of homepage url

Comment (TimWu007) [reply lib/v2/ycwb/index.js:70]: Thanks for reviewing. But for some top-level nodes, their articles may link to lower-nodes. E.g. node 18-广州, the node of its first article (https://news.ycwb.com/2023-02/28/content_51784570.htm) is 6030-广州>社会百态.

Comment (TimWu007) [reply lib/v2/ycwb/index.js:70]: @TonyRL Thanks for reviewing. But for some top-level nodes, their articles may link to lower-nodes. E.g. node 18-广州, the node of its first article (https://news.ycwb.com/2023-02/28/content_51784570.htm) is 6030-广州>社会百态.

Comment (TimWu007) [reply lib/v2/ycwb/index.js]: @TonyRL Hi, please review my latest commit. I've done of filtering node url from article page. www

File: lib/v2/ycwb/index.js [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+
+                nodeName = nodeName === '' ? item.nodeName : nodeName;
+
+                // Filter node link from content('.path a')
+                const children = content('.path').children('a');
+                for (let i = 0; i < children.length; i++) {
+                    if (content(children[i]).text() === nodeName && nodeLink === '') {
+                        nodeLink = content(children[i]).attr('href');
+                    }
+                }
```
Comment (TonyRL) [inline lib/v2/ycwb/index.js:62]:
```suggestion
                for (const child of children) {
                    if (content(child).text() === nodeName && nodeLink === '') {
                        nodeLink = content(child).attr('href');
                    }
                }
```
You can use [`.each()`](https://api.jquery.com/each/) as well

File: lib/v2/ycwb/maintainer.js [hunk 6]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
+    '/': ['TimWu007'],
```
Comment (TonyRL) [inline lib/v2/ycwb/maintainer.js:2]:
```suggestion
    '/:node': ['TimWu007'],
```
This should be the same as `lib/v2/ycwb/router.js`

File: lib/v2/ycwb/radar.js [hunk 7]
```diff
@@ -0,0 +1,12 @@
+module.exports = {
+    'xwlb.com.cn': {
+        _name: '羊城晚报金羊网',
+        '.': [
+            {
+                title: '新闻',
+                docs: 'https://docs.rsshub.app/traditional-media.html#yang-cheng-wan-bao',
```
Comment (TonyRL) [inline lib/v2/ycwb/radar.js:7]:
```suggestion
                docs: 'https://docs.rsshub.app/traditional-media.html#yang-cheng-wan-bao-jin-yang-wang',
```

**Pull request merged**